### ISSUE TITLE
Scroll view regains velocity as it scrolls

### ISSUE THREAD

**Issue by nt4f04uNd** [flutter/flutter#83632]
## Steps to Reproduce

Open and scroll license details in LicensePage

The issue is not present in the current gallery build, i.e. this is a regression

**Expected results:**

https://user-images.githubusercontent.com/39104740/120113922-6983f980-c185-11eb-92f6-7269e84f1ba2.mp4

**Actual results:**

https://user-images.githubusercontent.com/39104740/120113846-0befad00-c185-11eb-875b-9cdf2e8ffcc2.mp4

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(App());
}

class App extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'App',
      home: Scaffold(
        body: Center(
          child: Builder(
            builder: (context) {
              return ElevatedButton(
                child: Text('f'),
                onPressed: () {
                  showLicensePage(context: context);
                },
              );
            }
          )
        ),
      ),
    );
  }
}
```
</details>

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 2.3.0-17.0.pre.120, on Microsoft Windows [Version 10.0.19041.985], locale ru-RU)
    • Flutter version 2.3.0-17.0.pre.120 at C:\dev\src\flutter
    • Upstream repository git@github.com:nt4f04und/flutter.git
    • Framework revision c42ec8592f (2 days ago), 2021-05-28 22:34:02 -0400
    • Engine revision aca8aa2335
    • Dart version 2.14.0 (build 2.14.0-165.0.dev)

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.7.7)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.7.30621.155
    • Windows 10 SDK version 10.0.19041.0

[✓] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] IntelliJ IDEA Community Edition (version 2020.3)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2020.3.3
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.56.2)
    • VS Code at C:\Users\danya\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.22.0

[✓] Connected device (4 available)
    • sdk gphone x86 (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Windows (desktop)       • windows       • windows-x64    • Microsoft Windows [Version 10.0.19041.985]
    • Chrome (web)            • chrome        • web-javascript • Google Chrome 90.0.4430.212
    • Edge (web)              • edge          • web-javascript • Microsoft Edge 90.0.818.66

! Doctor found issues in 1 category.
```
</details>

**Comment by TahaTesser** [comment]
Hi @nt4f04uNd 
Thanks for filing the issue, I can reproduce the issue

Reproduced the issue using the following minimal code sample 

<details> 
<summary>minimal code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(App());
}

class App extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'App',
      home: Scaffold(
        body: Center(
          child: Builder(
            builder: (context) {
              return ElevatedButton(
                child: Text('f'),
                onPressed: () {
                  showLicensePage(context: context);
                },
              );
            }
          )
        ),
      ),
    );
  }
}

``` 

</details>

Reproduced on the following channels
|  `stable`  | `master` |
| :---: |:---: |
|  ❌  |   ❌  | 

Check `flutter doctor -v` outputs for each channel below

<details> 
<summary>flutter doctor -v</summary> 
	
```console
[✓] Flutter (Channel stable, 2.2.1, on Linux, locale en_US.UTF-8)
    • Flutter version 2.2.1 at /home/taha/Code/flutter_stable
    • Framework revision 02c026b03c (4 days ago), 2021-05-27 12:24:44 -0700
    • Engine revision 0fdb562ac8
    • Dart version 2.13.1

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /home/taha/Code/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /home/taha/Code/SDK
    • Java binary at: /home/taha/Code/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+0-b944-P17168821)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio (version 4.2)
    • Android Studio at /home/taha/Code/android-studio
    • Flutter plugin version 56.0.2
    • Dart plugin version 202.8488
    • Java version OpenJDK Runtime Environment (build 11.0.8+0-b944-P17168821)

[✓] VS Code (version 1.56.2)
    • VS Code at /usr/share/code
    • Flutter extension version 3.22.0

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Linux
    • Chrome (web)    • chrome • web-javascript • Google Chrome 91.0.4472.77

• No issues found!

```

```console
[✓] Flutter (Channel master, 2.3.0-17.0.pre.121, on Pop!_OS 20.04 LTS
    5.11.0-7614-generic, locale en_US.UTF-8)
    • Flutter version 2.3.0-17.0.pre.121 at /home/taha/Code/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 71e711547f (11 hours ago), 2021-05-30 19:31:03 -0700
    • Engine revision 73fe3dd31c
    • Dart version 2.14.0 (build 2.14.0-165.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /home/taha/Code/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /home/taha/Code/SDK
    • Java binary at: /home/taha/Code/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+0-b944-P17168821)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio (version 4.2)
    • Android Studio at /home/taha/Code/android-studio
    • Flutter plugin version 56.0.2
    • Dart plugin version 202.8488
    • Java version OpenJDK Runtime Environment (build 11.0.8+0-b944-P17168821)

[✓] VS Code (version 1.56.2)
    • VS Code at /usr/share/code
    • Flutter extension version 3.22.0

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Pop!_OS 20.04 LTS
      5.11.0-7614-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 91.0.4472.77

• No issues found!

```
		
</details>

✅ : No Issue ❌: has Issue

**Comment by goderbauer** [comment]
Can you describe what the issue is? From the video it's not clear to me what's broken.

**Comment by nt4f04uNd** [comment]
The fling scroll lasts too long

**Comment by ataberkw** [comment]
https://user-images.githubusercontent.com/10375421/120908375-559f3280-c672-11eb-9043-0f55a3bc674a.mov

It acts same on any scrollable content. When you slide it a bit, it gains unlogical velocity. Issue happens all of my project's last build.

```
[✓] Flutter (Channel stable, 2.2.1, on macOS 11.2.3 20D91 darwin-arm, locale en-TR)
    • Flutter version 2.2.1 at /Users/ataberkw/dev/flutter
    • Framework revision 02c026b03c (9 days ago), 2021-05-27 12:24:44 -0700
    • Engine revision 0fdb562ac8
    • Dart version 2.13.1

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/ataberkw/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] Connected device (3 available)
    • iPhone 12 (mobile)     • A17D4239-EA22-4C43-A5AB-B1BADAC65574 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • iPhone 12 Pro (mobile) • 8DF77AC8-96B4-408D-BCF3-976DFF424BB6 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • Chrome (web)           • chrome                               • web-javascript • Google
      Chrome 91.0.4472.77

! Doctor found issues in 1 category.
```

**Comment by mrcsilverfox** [comment]
> Screen.Recording.2021-06-06.at.02.52.45.mov 
> 
> It acts same on any scrollable content. When you slide it a bit, it gains unlogical velocity. Issue happens all of my project's last build.
> 
> Problem doesn't occur on a physical device.
> 
> ```
> [✓] Flutter (Channel stable, 2.2.1, on macOS 11.2.3 20D91 darwin-arm, locale en-TR)
>     • Flutter version 2.2.1 at /Users/ataberkw/dev/flutter
>     • Framework revision 02c026b03c (9 days ago), 2021-05-27 12:24:44 -0700
>     • Engine revision 0fdb562ac8
>     • Dart version 2.13.1
> 
> [!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
>     • Android SDK at /Users/ataberkw/Library/Android/sdk
>     • Platform android-30, build-tools 30.0.3
>     • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
>     • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
>     ✗ Android license status unknown.
>       Run `flutter doctor --android-licenses` to accept the SDK licenses.
>       See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
> 
> [✓] Xcode - develop for iOS and macOS
>     • Xcode at /Applications/Xcode.app/Contents/Developer
>     • Xcode 12.5, Build version 12E262
>     • CocoaPods version 1.10.1
> 
> [✓] Chrome - develop for the web
>     • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
> 
> [✓] Android Studio (version 4.2)
>     • Android Studio at /Applications/Android Studio.app/Contents
>     • Flutter plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/9212-flutter
>     • Dart plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/6351-dart
>     • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
> 
> [✓] Connected device (3 available)
>     • iPhone 12 (mobile)     • A17D4239-EA22-4C43-A5AB-B1BADAC65574 • ios            •
>       com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
>     • iPhone 12 Pro (mobile) • 8DF77AC8-96B4-408D-BCF3-976DFF424BB6 • ios            •
>       com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
>     • Chrome (web)           • chrome                               • web-javascript • Google
>       Chrome 91.0.4472.77
> 
> ! Doctor found issues in 1 category.
> ```

Me too, on MacBook Pro M1 as well

**Comment by ataberkw** [comment]
`Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.4 20F71 darwin-arm, locale
    en-TR)
[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[!] Android Studio
    ✗ Unable to find bundled Java version.
[✓] Android Studio (version 4.2)
[✓] Connected device (2 available)`

Still exists on same machine. All of my last builds have the issue. Is there any suggestion to solve it? Because I started to believe we are the only one who have this issue.

**Comment by csells** [comment]
dup https://github.com/flutter/flutter/issues/84197 found on 2.5 and 2.6

**Comment by Piinks** [comment]
I have bisected this, it was caused by https://github.com/flutter/flutter/pull/77497 cc/ @grouma 
This is strange because that change is intentionally updating the scrolling simulation to better match native Android.

**Comment by Piinks** [comment]
Chatted with @goderbauer and we are going to revert the change that caused this, unfortunately it has been in for a while so it can't be done automatically. I should be able to do this manually tomorrow and add some extra tests to prevent another regression while I am at it.

**Comment by csells** [comment]
thanks for tracking this down, @Piinks !

**Comment by ataberkw** [comment]
@Piinks  I'm so glad someone is interested in this issue. I don't know why are we the minority who experience this bug. I have this issue since 2.2.1 (not exact version this problem appeared). All of my simulator devices having the issue, not android and not real ios devices. This is issue makes iOS tests very hard.

Issue appears all of the builds I make on this mac. Even on the hello world app. Also it appears on any scrollable container as far I know.

https://user-images.githubusercontent.com/10375421/132781250-9ce753d4-f553-439b-b173-6c2924409762.mov


 ```
flutter doctor -v
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.4 20F71 darwin-arm, locale
    en-TR)
    • Flutter version 2.2.3 at /Users/ataberkw/dev/flutter
    • Framework revision f4abaa0735 (2 months ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/ataberkw/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] Connected device (3 available)
    • sdk gphone64 arm64 (mobile) • emulator-5554                        •
      android-arm64  • Android 11 (API 30) (emulator)
    • iPhone 12 (mobile)          • A17D4239-EA22-4C43-A5AB-B1BADAC65574 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • Chrome (web)                • chrome                               •
      web-javascript • Google Chrome 93.0.4577.63

! Doctor found issues in 2 categories.
```

**Comment by grouma** [comment]
> Chatted with @goderbauer and we are going to revert the change that caused this, unfortunately it has been in for a while so it can't be done automatically. I should be able to do this manually tomorrow and add some extra tests to prevent another regression while I am at it.

Is that the best course of action? My change fixes the behavior in real apps on real devices. Prior to my change, the scroll velocity would jump near the end of a scroll and would cause jitter. 

Can someone on Flutter not reproduce this locally? Ideally we fix this issue or work around the problem specifically in the simulator as to not regress real app behavior.

**Comment by goderbauer** [comment]
> Ideally we fix this issue or work around the problem specifically in the simulator as to not regress real app behavior.

My understanding was that the newly introduced problem affects real devices as well, not just simulators?

**Comment by Piinks** [comment]
Yes, @goderbauer, this is not restricted to device simulators, it can be reproduced on physical devices.

**Comment by SergejSachs** [comment]
Yes, it can we reproduced on real devices and simulators.

**Comment by grouma** [comment]
> > Ideally we fix this issue or work around the problem specifically in the simulator as to not regress real app behavior.
> 
> My understanding was that the newly introduced problem affects real devices as well, not just simulators?

I was going off this quote:

> All of my simulator devices having the issue, not android and not real ios devices

But digging through other duplicate issues it appears that this does impact real devices. A rollback may be the right course of action here but we'll reintroduce https://github.com/flutter/flutter/issues/16371 unfortunately.

**Comment by goderbauer** [comment]
> A rollback may be the right course of action here but we'll reintroduce #16371 unfortunately.

Agreed. #16371 needs to be fixed eventually, but for now I think we should roll this change back until we have time to investigate as this bug appears to be more annoying than #16371.

**Comment by Piinks** [comment]
Thanks @grouma & @goderbauer. I am taking a look at that hard coded value just in case it does solve the issue. Otherwise I'll start the revert. 

Just for reference here are the simulator/real device repros:

https://user-images.githubusercontent.com/16964204/132889493-9177d6bd-bbe2-4e3b-b936-ef8bc90a4298.mov


https://user-images.githubusercontent.com/16964204/132890019-5691317c-6049-43a3-b178-25aa2e9d7fc9.MOV

**Comment by Piinks** [comment]
Update:

I checked out the hardcoded value @grouma suggested. I tried a couple of options, setting the actual ppi of the device among them. `_decelerationForFriction` does not appear to be where this issue is coming from, because it is nearly the same as before the change: https://github.com/flutter/flutter/pull/77497/files

Before the change it was `friction * 61774.04968`
and the change made it `9.80665 * 39.37 * friction * 1.0 * 160.0` which is `61774.04967`.
It's the same as before.
This makes me think the issue is elsewhere.
 I'll start to roll this back after lunch.
